addons/mail/models/discuss/mail_guest.py

```diff
@@ -32,8 +32,8 @@ def wrapper(self, *args, **kwargs):
                 guest._update_timezone(timezone)
         if guest:
             req.update_context(guest=guest)
-            if hasattr(self, "env"):
-                self.env.context = {**self.env.context, "guest": guest}
+            if isinstance(self, models.BaseModel):
```

Comment:
It doesn't sound correct and I don't understand how tests are green. `self` can be a Controller too (which have a env property), right ? 
Or I miss something ?